Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs

```diff
@@ -0,0 +1,317 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    [RequireComponent(typeof(Camera))]
+    public class MixedRealityInputModule : StandaloneInputModule, IMixedRealityPointerHandler, IMixedRealitySourceStateHandler
+    {
+#if UNITY_EDITOR
+        [UnityEditor.CustomEditor(typeof(MixedRealityInputModule))]
+        public class Editor : UnityEditor.Editor
+        {
+            public override void OnInspectorGUI()
+            {
+                MixedRealityInputModule t = (MixedRealityInputModule)target;
+
+                base.OnInspectorGUI();
+
+                if (Application.isPlaying && t.RaycastCamera != null)
+                {
+                    foreach (var pointerData in t.pointerDataToUpdate)
+                    {
+                        IMixedRealityPointer pointer = pointerData.Value.pointer;
+                        if (pointer.Rays != null && pointer.Rays.Length > 0)
+                        {
+                            t.RaycastCamera.transform.position = pointer.Rays[0].Origin;
+                            t.RaycastCamera.transform.rotation = Quaternion.LookRotation(pointer.Rays[0].Direction, Vector3.up);
+
+                            t.RaycastCamera.Render();
+
+                            GUILayout.Label(pointer.PointerName);
+                            GUILayout.Label(pointer.ToString());
+                            GUILayout.Label(pointer.PointerId.ToString());
+                            GUILayout.Label(t.RaycastCamera.targetTexture);
+                        }
+                    }
+                }
+            }
+
+            public override bool RequiresConstantRepaint()
+            {
+                return true;
+            }
+        }
+#endif
+
+        // TODO: Make sure Unity UI input capturing participates in MRTK focus locking correctly.
+        // TODO: Consider simulating touch events rather than mouse events for PokePointer. What are the tradeoffs?
+
```

Comment:
Nit: It would probably be better to have these tracked as issues rather than TODOs (or having TODOs that point to filed issues). Unless we plan on resolving these in the next week, these will likely get forgotten and then not be helpful in the future. It's unfortunately the fate of all TODOs for people who have too much todo.

Reply:
This was part of the original MixedRealityInputModule. Will create a ticket.

Reply:
https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/3759